test/functional/test_framework/script.py

```diff
@@ -67,18 +67,22 @@ def encode_op_pushdata(d):
     @staticmethod
     def encode_op_n(n):
         """Encode a small integer op, returning an opcode"""
-        if not (0 <= n <= 16):
-            raise ValueError('Integer must be in range 0 <= n <= 16, got %d' % n)
+        if not (-1 <= n <= 16):
+            raise ValueError('Integer must be in range -1 <= n <= 16, got %d' % n)
 
         if n == 0:
             return OP_0
+        if n == -1:
+            return OP_1NEGATE
         else:
             return CScriptOp(OP_1 + n - 1)
 
     def decode_op_n(self):
         """Decode a small integer opcode, returning an integer"""
         if self == OP_0:
             return 0
+        if self == OP_1NEGATE:
+            return -1
 
         if not (self == OP_0 or OP_1 <= self <= OP_16):
```

Comment:
Checking `OP_0` is unnecessary here, but now that it's not checking `OP_1NEGATE` it's inconsistently unnecessary. How about using `is_small_int` to avoid code duplication?

```python
def decode_op_n(self):
    """Decode a small integer opcode, returning an integer"""
    if not self.is_small_int():
        raise ValueError('op %r is not an OP_N' % self)

    if self == OP_0:
        return 0
    elif self == OP_1NEGATE:
        return -1
    else:
        return int(self - OP_1 + 1)
```

Reply:
Done in 6cf08dd2d532766f1b7063f40e7ab25f8ee67e45